drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -425,8 +632,14 @@ impl Transceivers {
 
             // get the relevant data from the correct FPGA
             let local_data = match port_loc.controller {
-                FpgaController::Left => &f0,
-                FpgaController::Right => &f1,
+                FpgaController::Left => match ldata {
```

Comment:
I can't apply a suggestion to the correct block, but you could rewrite this section of code as
```rust
            // get the relevant data from the correct FPGA
            let local_data = match port_loc.controller {
                FpgaController::Left => ldata,
                FpgaController::Right => rdata,
            };
            let Some(local_data) = local_data else { continue };
```